### PR TITLE
feat(cli): add retry to cache operations

### PR DESCRIPTION
**Pull request by roi-orca**
## Related issues
- Close #XXX

## Related PRs
- [ ] #XXX
- [ ] #YYY

Remove this section if you don't have related PRs.

## Checklist
- [ ] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [ ] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/cache/remote.go [hunk 1]
```diff
@@ -38,7 +39,12 @@ func NewRemoteCache(url string, customHeaders http.Header, insecure bool) cache.
 
 // PutArtifact sends artifact to remote client
 func (c RemoteCache) PutArtifact(imageID string, artifactInfo types.ArtifactInfo) error {
-	_, err := c.client.PutArtifact(c.ctx, rpc.ConvertToRPCArtifactInfo(imageID, artifactInfo))
+	var _ *emptypb.Empty
```
Comment (knqyf263) [inline pkg/cache/remote.go:42]: What is this for?

Comment (roi-orca) [reply pkg/cache/remote.go:42]: The response that is not in used. What should I do instead?

Comment (knqyf263) [reply pkg/cache/remote.go:42]: What's going to happen if we just remove this line?

Comment (roi-orca) [reply pkg/cache/remote.go:42]: I need to add it to the inner close function

Comment (knqyf263) [reply pkg/cache/remote.go:42]: Did you see any errors?

Comment (roi-orca) [reply pkg/cache/remote.go]: updated

**Review by knqyf263 — APPROVED** [review]

**Comment by knqyf263** [comment]
@roi-orca Please update the PR description next time.

**Comment by roi-orca** [comment]
@knqyf263 can you merge the pr? thanks

**Pull request merged**